[Bug] JAXB module can't be well loaded in JDK11 environment

### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.


### Ray Component

Ray Core

### What happened + What you expected to happen

## Bug 
Following exception will be thrown out in worker process when creating instance of class `javax.xml.bind.JAXBContext`.
```
Caused by: java.lang.ClassNotFoundException: javax.activation.DataSource
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 43 more
```

## Inforamtion
As we known JAXB module was removed from JDK9+, so following maven dependencies are needed to be packaged into user jars:
```
<dependency>
    <groupId>javax.activation</groupId>
    <artifactId>activation</artifactId>
</dependency>
<dependency>
    <groupId>javax.xml.bind</groupId>
    <artifactId>jaxb-api</artifactId>
</dependency>
<dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-impl</artifactId>
</dependency>
<dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-core</artifactId>
</dependency>
```
But `ray_dist.jar` in ray installed directory `${python-site-packages-dir}/ray/jars` only contains dependencies above except `activation`. The lost dependency jar can't be loaded well even user packages it into worker's running jar or adds it into worker's classpath, because `JAXBContext` in jar `jaxb-api` has been loaded by `AppClassLoader` of ray worker by default. Hence, class `javax.activation.DataSource` in jar `activation` is referenced to be loaded by `AppClassLoader` when creating instance of `JAXBContext` for the reason of implementation of `JAXBContext.newInstance` and `ClassNotFoundException` will be thrown !

## Solution
Packaging maven dependency `activation` into `ray_dist.jar` will fix this problem.

### Versions / Dependencies

Ray：all
Java：11.0.12


### Reproduction script

```java
import javax.xml.bind.JAXBContext;

// call ray worker to do so
JAXBContext context = JAXBContext.newInstance(Object.class);
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

--- a/java/BUILD.bazel
+++ b/java/BUILD.bazel
@@ -48,6 +48,7 @@ define_java_module(
         "@maven//:com_lmax_disruptor",
         "@maven//:com_sun_xml_bind_jaxb_core",
         "@maven//:com_sun_xml_bind_jaxb_impl",
+        "@maven//:javax_activation_activation",
         "@maven//:javax_xml_bind_jaxb_api",
         "@maven//:org_apache_logging_log4j_log4j_api",
         "@maven//:org_apache_logging_log4j_log4j_core",

--- a/java/dependencies.bzl
+++ b/java/dependencies.bzl
@@ -15,6 +15,7 @@ def gen_java_deps():
             "commons-io:commons-io:2.7",
             "de.ruedigermoeller:fst:2.57",
             "javax.xml.bind:jaxb-api:2.3.0",
+            "javax.activation:activation:1.1.1",
             "org.apache.commons:commons-lang3:3.4",
             "org.msgpack:msgpack-core:0.8.20",
             "org.ow2.asm:asm:6.0",

